src/ingestors/transient-base/index.ts

```diff
@@ -0,0 +1,91 @@
+import { MintInstructionType, MintTemplate } from '../../lib/types/mint-template';
+import {
+  getTransientBaseMintByAddressAndChain,
+  getTransientBaseMintByURL,
+  transientSupports,
+} from './offchain-metadata';
+
+import { MintTemplateBuilder } from '../../lib/builder/mint-template-builder';
+import { TRANSIENT_BASE_ABI } from './abi';
+
+export class TransientIngestor implements MintIngestor {
+  configuration = {
+    supportsContractIsExpensive: true,
+  };
+
+  async supportsUrl(resources: MintIngestorResources, url: string): Promise<boolean> {
+    if (new URL(url).hostname !== 'www.transient.xyz') {
+      return false;
+    }
+    const { chainId, contractAddress } = await getTransientBaseMintByURL(resources, url);
+    return !!chainId && !!contractAddress;
+  }
+
+  async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+    const { chainId, contractAddress } = contract;
+    if (!chainId || !contractAddress) {
+      return false;
+    }
+    return await transientSupports(contract, resources.fetcher);
+  }
+
+  async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+    const isCompatible = await this.supportsUrl(resources, url);
+    if (!isCompatible) {
+      throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+    }
+
+    const { chainId, contractAddress } = await getTransientBaseMintByURL(resources, url);
+
+    if (!chainId || !contractAddress) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    return this.createMintForContract(resources, { chainId, contractAddress, url });
+  }
+
+  async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+    const { chainId, contractAddress } = contract;
+    if (!chainId || !contractAddress) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('Transient');
+
+    if (contract.url) {
+      mintBuilder.setMarketingUrl(contract.url);
+    }
+
+    // asides name and image no other metadata we need is onchain so we can just use the offchain metadata
+    const { name, image, description, priceInWei, mintAddress, public_sale_start_at, public_sale_end_at } =
+      await getTransientBaseMintByAddressAndChain(resources, contract.chainId, contract.contractAddress);
+
+    mintBuilder.setName(name).setDescription(description).setFeaturedImageUrl(image);
+
+    mintBuilder.setMintInstructions({
+      chainId,
+      contractAddress,
+      mintAddress,
+      contractMethod: 'purchase',
+      contractParams: '["address", 1, 1, address]',
```

Comment:
Looking at the ABI above, is this right?

https://github.com/floornfts/mobile-minting/pull/27/files#diff-7b9dec2faf20a75b53006dc8f4835e4f6d5e131c7287e038cb46cccd657bb7daR11

Are the last properties optional? And are `nftAddress` and `recipient` supposed to both be the recipient's address?

Reply:
```
 // nftAddress	address	0x8F7F96eEc76c882C327a1b927D04C87031ED5481
// 1	tokenId	uint256	1
// 2	recipient	address	0xcA4700293Bd3b48E7051116635C82546F17a61e3
// 3	numberToMint	uint256	1
// 4	presaleNumberCanMint	uint256	0
// 5	proof	bytes32[]
```

The above is a sample contract interaction pulled from the explorer for a successful purchase on transient

Reply:
The nftAddress should be the address of the nft
and yes recipient address should be address of recipient. ABI says this allows for buying for others

Reply:
I'm a bit unsure of what `contractParams: '["address", 1, 1, address]',` should be atm

Reply:
I set it to
`["${contract.contractAddress}", 1, address, 1]`,

I believe it represents the structure better now